### DISCUSSION TITLE
Problem with Book Instructions for Adding Admin Login

### DISCUSSION THREAD

**Discussion by gdgray** [symfony/symfony#49287] [Symfony Book: The Fast Track] [answered]
Chapter 15-Security
    
Where is says: "Select App\Entity\Admin and then choose whatever you like as a password and run the following command to generate the password hash ..."
    
What does it mean by "Select App\Entity\Admin"? Are we supposed edit something in that file?

We've followed the instructions but can't login. The server log shows the following:

[Application] [2023-02-07T13:07:19.523375-06:00] security.DEBUG: Checking for authenticator support. {"firewall_name":"main","authenticators":1} []
[Application] [2023-02-07T13:07:19.523413-06:00] security.DEBUG: Checking support on authenticator. {"firewall_name":"main","authenticator":"App\\Security\\AppAuthenticator"} []
[Application] [2023-02-07T13:07:19.523436-06:00] security.DEBUG: Authenticator does not support the request. {"firewall_name":"main","authenticator":"App\\Security\\AppAuthenticator"} []

I can certainly post more but the answer to the "Select App\Entity\Admin" question might reveal an answer.

**Comment by MatTheCat** [comment, marked as answer]
I guess it should read

> Select `App\Entity\Admin` and then choose whatever you like as a password [**when running**] the following command to generate the password hash:
> ```shell
> symfony console security:hash-password
> ```

Would that make sense to you?

**Comment by gdgray** [comment]
Not really (but thanks for the reply). I'm not sure how I would "Select App\Entity\Admin". What is this "select" process?

**Reply by MatTheCat** [reply]
Oh yeah the `security:hash-password` won’t ask you the user class since you only have `App\Entity\Admin` at this point. But if you had defined different password hashers for different user classes you would had to “select App\Entity\Admin”.

**Reply by gdgray** [reply]
Thanks for the clarification.
